feat(data-exploration): "saved" queries

mariusandra opened the pull request:
## Problem

It's hard to understand that by adding a column like `count()`, you can convert the "Event Explorer" scene into a powerful data table.

## Changes

Adds a few saved queries to get people exploring: (gif below)

~Also rearranges the filters above the table to better suit this button.~

Everything's still flagged 😅 

## Out of scope

I'd love for there to be a way to save your own queries and also set a new default for the team... but... let's discuss this. We're awkwardly close to recreating saved insights 🤔 

## How did you test this code?

Clicked around in the table.

mariusandra committed:
feat(data-exploration): rename to "Event Explorer"

mariusandra committed:
feat(data-exploration): saved queries in data table

mariusandra committed:
Merge branch 'master' into saved-queries

Twixes commented:
This is increasingly becoming an insight… I also have a suspicion that the "hard to understand that by adding a column you can convert this into a powerful data table" problem would be kind of solved by separating the "Live Events" and "Data Table" use cases.

mariusandra commented:
> I also have a suspicion that the "hard to understand that by adding a column you can convert this into a powerful data table" problem would be kind of solved by separating the "Live Events" and "Data Table" use cases.

In a "Data Table" usecase you'd still have the same problem: how to go from a list view to an aggregate view. Sometimes you want to see every event, sometimes you want to see an aggregation. Said differently, assuming no 

> This is increasingly becoming an insight… 

Yeah, agreed. 

However, just slapping this entire table under "insights" right now seems premature. We can always move this around in a few PRs, but database changes are hard. So let's think it through. I feel there are some missing pieces, and work soon enough will wrap all of them up:

- There's an issue about [renaming "Insights" into "Product Analytics"](https://github.com/PostHog/posthog/issues/13742)
- [PostHog 3000](https://github.com/orgs/PostHog/projects/78/views/1) has a "notebook mode" design idea. When it'll be built is unknowon.
- We'd like data exploration nodes on dashboards. I wouldn't want to now add a new insight type, but this should be just a different node in a unified system.
- It might be better (or at least interesting) to start an exploration from the "data table" view, and convert to a visualisation automatically when you have any aggregations. Plus add a horizontal time axis if you include the timestamp.

mariusandra committed:
fix test

mariusandra committed:
put columns back

mariusandra committed:
cleanup

mariusandra commented:
I fixed a test bug, and moved the table header back to have the smallest possible change in this PR:

![2023-01-23 14 49 35](https://user-images.githubusercontent.com/53387/214056152-f5601860-9d4c-4842-9175-548eabddcc49.gif)

mariusandra commented on frontend/src/queries/nodes/DataNode/dataNodeLogic.ts:
```diff
@@ -203,6 +203,8 @@ export const dataNodeLogic = kea<dataNodeLogicType>([
                             if (firstTimestamp) {
                                 const nextQuery: EventsQuery = { ...query, after: firstTimestamp }
                                 return nextQuery
+                            } else {
+                                return query
```
This fixes a bug: before, if there were no entries in the initial response, the "automatically load more events" button would not be shown.

mariusandra committed:
räägime inglise keeles

mariusandra committed:
show no columns if loading data of different type

mariusandra commented:
Meh. I didn't like how the page flickered when changing queries (screencast above). Updated now (screencast below) and ready for a re-review.

![2023-01-23 15 22 32](https://user-images.githubusercontent.com/53387/214063421-fa39667c-ab59-4a47-a6a3-f2ecb72a8b73.gif)

mariusandra committed:
default to off for showing saved queries

pauldambra commented on frontend/src/queries/schema.ts:
```diff
@@ -204,6 +204,8 @@ export interface DataTableNode extends Node {
     showElapsedTime?: boolean
     /** Show a button to configure the table's columns if possible */
     showColumnConfigurator?: boolean
+    /** Shows a list of saved queries */
```
nit: you could live without the comment here 😅

pauldambra reviewed (approved):
🚢

mariusandra committed:
Merge branch 'master' into saved-queries

mariusandra commented:
I'm going to use @pauldambra 's ✅ as an excuse to move fast and iterate
